pkg/execution/executor/executor.go

```diff
@@ -1229,7 +1390,9 @@ func (e *executor) handleGeneratorWaitForEvent(ctx context.Context, gen state.Ge
 	// the pause so this race will conclude by calling the function once, as only
 	// one thread can lease and consume a pause;  the other will find that the
 	// pause is no longer available and return.
+	jobID := fmt.Sprintf("%s-%s-%s", item.Identifier.IdempotencyKey(), gen.ID, "wait")
 	err = e.queue.Enqueue(ctx, queue.Item{
+		JobID:       &jobID,
```

Comment:
Why was this added?  I understand it, but an curious given waitForEvent has existed since day zero.

Reply:
Hm I think this might have been creating some duplicate calls after a `waitForEvent` after I'd added `invoke`. https://github.com/inngest/inngest/commit/30c7b8db4bc98914b6d7374b9409c3868c0f8d1b

I can pull this out and debug a touch - a comment would be good if it's the case.

Reply:
@tonyhb atm, because `waitForEvent` has no `JobID`, it looks like it uses an empty string `""` for idempotency. When that happens, it looks like we instead use a generated ID for idempotency, which is obviously different every time.

If two tools lack setting a `JobID` then parallelism can cause us to re-enqueue the same steps again and again because idempotency is basically removed.

In fact, while this is mostly a fix for adding a new tool, this would definitely cause problems with parallelism and `waitForEvent` anyway. Setting this now won't cause any more damage than it being unset currently. I'll raise a separate PR to get that fix in.

Reply:
@tonyhb Have added #859 for this.